### DISCUSSION TITLE
Physics3D how to set fixedRotation such that my character 3D model will not rotate after collision?

### DISCUSSION THREAD

**Discussion by cchuicchui** [axmolengine/axmol#864] [Q&A] [answered]
In Physics3D how to set fixedRotation such that my character 3D model will not rotate after collision?

I always want my 3D character to stand up but now when my character hits some other objects in the game it rotates incorrectly.
<img width="240" alt="Screenshot 2022-09-25 at 5 06 41 PM" src="https://user-images.githubusercontent.com/39337211/192136221-ffcbbb85-38b1-4abe-801b-3ca4de9f1a84.png">

I want my 3D character to bound back but it won't fall down.

In cocos creator, there is a feature called fixedRotation but I cannot find this in axys:

          When doing platform jumping games, you usually don't want the rotation property 
          of the main character to be added to the physics simulation, because it
          will cause the main character to fall over during the movement, so you
          can set fixedRotation of the rigid body to true to fix the rotation, the code
          example is as follows:

                    rigidbody.fixedRotation = true;

**Comment by aismann** [comment]
@cchuicchui 
Can you make an (github) or cpp-tests example?

**Comment by rh101** [comment, marked as answer]
Read this and see if it helps:

https://pybullet.org/Bullet/phpBB3/viewtopic.php?p=12113&sid=57b2b4c3ef51ca26decd572b4f15bb94#p12113
https://gamedev.stackexchange.com/questions/22319/how-to-disable-y-axis-movement-in-the-bullet-physics-engine

You'll most likely need to call this method:

`void setAngularFactor(const cocos2d::Vec3& angFac);`

**Reply by cchuicchui** [reply]
The following code works like a charm!

          rigidBody->setAngularFactor(Vec3(0, 0, 0));
